### PR TITLE
Fixes some issues with the pool

### PR DESCRIPTION
**Pull request by tortellinitony**
- All carbon mobs will now be able to fall in, etc.
- All mobs will make water noises when moving, and will be washed.
- Fixed an issue where players could not move after diving.
- Added a slight bobbing effect to mobs in the pool.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: hippiestation/code/modules/pool/pool_main.dm [hunk 1]
```diff
@@ -406,14 +409,14 @@
 			return
 		else
 			playsound(src, 'hippiestation/sound/effects/watersplash.ogg', 8, 1, 1)
-			src.splashed = 1
+			src.splashed = TRUE
```
Comment (ghost) [inline hippiestation/code/modules/pool/pool_main.dm:412]: isn't src implicit here?

File: hippiestation/code/modules/pool/pool_main.dm [hunk 2]
```diff
@@ -406,14 +409,14 @@
...
 			var/obj/effect/splash/S = new /obj/effect/splash(user.loc)
 			animate(S, alpha = 0,  time = 8)
 			S.Move(src)
 			spawn(20)
 				qdel(S)
 				spawn(5)
-					src.splashed = 0 //Otherwise, infinite splash party.
+					src.splashed = FALSE //Otherwise, infinite splash party.
```
Comment (ghost) [inline hippiestation/code/modules/pool/pool_main.dm:419]: and here?

File: hippiestation/code/modules/pool/pool_main.dm [hunk 3]
```diff
@@ -184,67 +184,70 @@
...
-	else
-		if(ishuman(A))
-			var/mob/living/carbon/human/H = A
-			H.adjustStaminaLoss(1)
-			wash_mob(H)
-			if(H.swimming == 1)
-				playsound(src, pick('hippiestation/sound/effects/water_wade1.ogg','hippiestation/sound/effects/water_wade2.ogg','hippiestation/sound/effects/water_wade3.ogg','hippiestation/sound/effects/water_wade4.ogg'), 20, 1)
+	if(ismob(A))
+		var/mob/living/M = A
+		src.filled ? wash_mob(M) : null
```
Comment (ghost) [inline hippiestation/code/modules/pool/pool_main.dm:200]: and here?

**Review by ghost — DISMISSED** [review]

File: hippiestation/code/modules/pool/pool_main.dm (hunk 3 above)
Comment (tortellinitony) [reply hippiestation/code/modules/pool/pool_main.dm:200]: some people like to have src so it's easier to read

**Pull request merged**